### PR TITLE
[OSGi] Load jclouds bundles in Karaf

### PR DESCRIPTION
**Pull request by neykov**
Needed to break the `locations-jclouds` dependency on `googlecompute` as it doesn't have an osgi bundle in `1.9.2`.

With the change I am able to deploy `MachineEntity` to Amazon, Softlayer, Bluebox.

Depends on (but not based on):
  * #55
  * #65 
  * #66

### CODE REVIEW

**Comment by neykov** [comment]
Once tasks started piling up started getting the following error in the logs:

```
2016-03-21 17:33:11,450 | INFO  | tp1904918854-187 | LoggingInInterceptor             | 107 - org.apache.cxf.cxf-core - 3.1.4 | Inbound Message
----------------------------
ID: 557
Address: http://localhost:8081/v1/activities/vTzq6haF
Http-Method: GET
Content-Type: 
Headers: {Accept=[application/json, text/javascript, */*; q=0.01], accept-encoding=[gzip, deflate, sdch], Accept-Language=[en-US,en;q=0.8], Authorization=[Basic YWRtaW46YWRtaW4=], connection=[keep-alive], Content-Type=[null], Host=[localhost:8081], Referer=[http://localhost:8081/], User-Agent=[Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/49.0.2623.87 Safari/537.36], X-Requested-With=[XMLHttpRequest]}
--------------------------------------
2016-03-21 17:33:11,668 | ERROR | tp1904918854-187 | JAXRSUtils                       | 111 - org.apache.cxf.cxf-rt-frontend-jaxrs - 3.1.4 | Problem with writing the data, class org.apache.brooklyn.rest.domain.TaskSummary, ContentType: application/json
2016-03-21 17:33:11,671 | WARN  | tp1904918854-187 | PhaseInterceptorChain            | 107 - org.apache.cxf.cxf-core - 3.1.4 | Interceptor for {http://resources.rest.brooklyn.apache.org/}AccessResource has thrown exception, unwinding now
org.apache.cxf.interceptor.Fault: com/fasterxml/jackson/databind/JsonMappingException$Reference
        at org.apache.cxf.jaxrs.interceptor.JAXRSOutInterceptor.handleWriteException(JAXRSOutInterceptor.java:382)[111:org.apache.cxf.cxf-rt-frontend-jaxrs:3.1.4]
        at org.apache.cxf.jaxrs.interceptor.JAXRSOutInterceptor.serializeMessage(JAXRSOutInterceptor.java:257)[111:org.apache.cxf.cxf-rt-frontend-jaxrs:3.1.4]
        at org.apache.cxf.jaxrs.interceptor.JAXRSOutInterceptor.handleWriteException(JAXRSOutInterceptor.java:384)[111:org.apache.cxf.cxf-rt-frontend-jaxrs:3.1.4]
        at org.apache.cxf.jaxrs.interceptor.JAXRSOutInterceptor.serializeMessage(JAXRSOutInterceptor.java:257)[111:org.apache.cxf.cxf-rt-frontend-jaxrs:3.1.4]
        at org.apache.cxf.jaxrs.interceptor.JAXRSOutInterceptor.processResponse(JAXRSOutInterceptor.java:118)[111:org.apache.cxf.cxf-rt-frontend-jaxrs:3.1.4]
        at org.apache.cxf.jaxrs.interceptor.JAXRSOutInterceptor.handleMessage(JAXRSOutInterceptor.java:81)[111:org.apache.cxf.cxf-rt-frontend-jaxrs:3.1.4]
        at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:308)[107:org.apache.cxf.cxf-core:3.1.4]
        at org.apache.cxf.interceptor.OutgoingChainInterceptor.handleMessage(OutgoingChainInterceptor.java:83)[107:org.apache.cxf.cxf-core:3.1.4]
        at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:308)[107:org.apache.cxf.cxf-core:3.1.4]
        at org.apache.cxf.transport.ChainInitiationObserver.onMessage(ChainInitiationObserver.java:121)[107:org.apache.cxf.cxf-core:3.1.4]
        at org.apache.cxf.transport.http.AbstractHTTPDestination.invoke(AbstractHTTPDestination.java:251)[119:org.apache.cxf.cxf-rt-transports-http:3.1.4]
        at org.apache.cxf.transport.servlet.ServletController.invokeDestination(ServletController.java:234)[119:org.apache.cxf.cxf-rt-transports-http:3.1.4]
        at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:208)[119:org.apache.cxf.cxf-rt-transports-http:3.1.4]
        at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:160)[119:org.apache.cxf.cxf-rt-transports-http:3.1.4]
        at org.apache.cxf.transport.servlet.CXFNonSpringServlet.invoke(CXFNonSpringServlet.java:180)[119:org.apache.cxf.cxf-rt-transports-http:3.1.4]
        at org.apache.cxf.transport.servlet.AbstractHTTPServlet.handleRequest(AbstractHTTPServlet.java:293)[119:org.apache.cxf.cxf-rt-transports-http:3.1.4]
        at org.apache.cxf.transport.servlet.AbstractHTTPServlet.doGet(AbstractHTTPServlet.java:217)[119:org.apache.cxf.cxf-rt-transports-http:3.1.4]
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:687)[50:javax.servlet-api:3.1.0]
        at org.apache.cxf.transport.servlet.AbstractHTTPServlet.service(AbstractHTTPServlet.java:268)[119:org.apache.cxf.cxf-rt-transports-http:3.1.4]
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:808)[190:org.eclipse.jetty.servlet:9.2.10.v20150310]
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:587)[190:org.eclipse.jetty.servlet:9.2.10.v20150310]
        at org.ops4j.pax.web.service.jetty.internal.HttpServiceServletHandler.doHandle(HttpServiceServletHandler.java:70)[211:org.ops4j.pax.web.pax-web-jetty:4.2.3]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)[189:org.eclipse.jetty.server:9.2.10.v20150310]
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:577)[188:org.eclipse.jetty.security:9.2.10.v20150310]
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:223)[189:org.eclipse.jetty.server:9.2.10.v20150310]
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127)[189:org.eclipse.jetty.server:9.2.10.v20150310]
        at org.ops4j.pax.web.service.jetty.internal.HttpServiceContext.doHandle(HttpServiceContext.java:271)[211:org.ops4j.pax.web.pax-web-jetty:4.2.3]
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515)[190:org.eclipse.jetty.servlet:9.2.10.v20150310]
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)[189:org.eclipse.jetty.server:9.2.10.v20150310]
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061)[189:org.eclipse.jetty.server:9.2.10.v20150310]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)[189:org.eclipse.jetty.server:9.2.10.v20150310]
        at org.ops4j.pax.web.service.jetty.internal.JettyServerHandlerCollection.handle(JettyServerHandlerCollection.java:80)[211:org.ops4j.pax.web.pax-web-jetty:4.2.3]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)[189:org.eclipse.jetty.server:9.2.10.v20150310]
        at org.eclipse.jetty.server.Server.handle(Server.java:497)[189:org.eclipse.jetty.server:9.2.10.v20150310]
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:310)[189:org.eclipse.jetty.server:9.2.10.v20150310]
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257)[189:org.eclipse.jetty.server:9.2.10.v20150310]
        at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:540)[181:org.eclipse.jetty.io:9.2.10.v20150310]
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635)[192:org.eclipse.jetty.util:9.2.10.v20150310]
        at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555)[192:org.eclipse.jetty.util:9.2.10.v20150310]
        at java.lang.Thread.run(Thread.java:745)[:1.7.0_80]
Caused by: java.lang.NoClassDefFoundError: com/fasterxml/jackson/databind/JsonMappingException$Reference
        at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:177)[21:com.fasterxml.jackson.core.jackson-databind:2.4.5]
        at com.fasterxml.jackson.databind.ser.std.StdSerializer.wrapAndThrow(StdSerializer.java:187)[21:com.fasterxml.jackson.core.jackson-databind:2.4.5]
        at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:647)[21:com.fasterxml.jackson.core.jackson-databind:2.4.5]
        at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:152)[21:com.fasterxml.jackson.core.jackson-databind:2.4.5]
        at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:128)[21:com.fasterxml.jackson.core.jackson-databind:2.4.5]
        at com.fasterxml.jackson.databind.ObjectWriter.writeValue(ObjectWriter.java:602)[21:com.fasterxml.jackson.core.jackson-databind:2.4.5]
        at com.fasterxml.jackson.jaxrs.base.ProviderBase.writeTo(ProviderBase.java:648)
        at org.apache.cxf.jaxrs.utils.JAXRSUtils.writeMessageBody(JAXRSUtils.java:1378)[111:org.apache.cxf.cxf-rt-frontend-jaxrs:3.1.4]
        at org.apache.cxf.jaxrs.interceptor.JAXRSOutInterceptor.serializeMessage(JAXRSOutInterceptor.java:235)[111:org.apache.cxf.cxf-rt-frontend-jaxrs:3.1.4]
        ... 38 more
2016-03-21 17:33:11,672 | INFO  | tp1904918854-187 | LoggingOutInterceptor            | 107 - org.apache.cxf.cxf-core - 3.1.4 | Outbound Message
---------------------------
ID: 557
Response-Code: 500
Content-Type: application/json
Headers: {Content-Type=[application/json], Cache-Control=[no-cache, no-store], Pragma=[no-cache], Expires=[0]}
Messages: (message truncated to 49152 bytes)

Payload: {"id":"vTzq6haF","displayName":"provisioning (aws-ec2)","description":"","entityId":"XVIqFcDO","entityDisplayName":"MachineEntity:XVIq","tags":["org.apache.brooklyn.core.mgmt.BrooklynTaskTags$Entitle
```

File: karaf/features/src/main/feature/feature.xml [hunk 1]
```diff
@@ -16,12 +16,17 @@
...
+          name="org.apache.brooklyn-${project.version}"
+          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+          xsi:noNamespaceSchemaLocation="http://karaf.apache.org/xmlns/features/v1.2.0">
 
     <repository>mvn:org.apache.karaf.features/standard/${karaf.version}/xml/features</repository>
     <repository>mvn:org.apache.karaf.features/enterprise/${karaf.version}/xml/features</repository>
     <repository>mvn:org.apache.karaf.features/spring/${karaf.version}/xml/features</repository>
 
+    <!-- TODO how to add central.maven.org repo? -->
+    <repository>http://central.maven.org/maven2/org/apache/jclouds/karaf/jclouds-karaf/1.9.2/jclouds-karaf-1.9.2-features.xml</repository>
```
Comment (neykov) [inline karaf/features/src/main/feature/feature.xml]: /cc @geomacy @m4rkmckenna @CMoH 
Any idea how I can avoid using the absolute URL? Using a `mvn:` looks in `repo.apache.org` which is a 404.

**Comment by neykov** [comment]
Blueprint used to test:
```
brooklyn.catalog:
  items:
  - id: machine
    libraries:
    - name: org.apache.brooklyn.software-base
      version: 0.9.0.SNAPSHOT
    item:
      type: org.apache.brooklyn.entity.machine.MachineEntity
```
```
location: aws-ec2
services:
- type: machine
```

File: karaf/features/src/main/feature/feature.xml (hunk 1 above)
Comment (m4rkmckenna) [inline karaf/features/src/main/feature/feature.xml]: Does this work?
```
<repository>mvn:org.apache.jclouds/jclouds-karaf/1.9.2/xml/features</repository>
```

Im guess the feature file you are referencing is this one https://github.com/jclouds/jclouds-karaf/blob/master/feature/pom.xml

Comment (neykov) [inline karaf/features/src/main/feature/feature.xml]: Yes, that's the one. Unfortunately it's not able to find the jar file when using the `mvn` protocol.
Looking at the debug output it's trying to download from:
  * http://svn.apache.org/repos/asf/servicemix/m2-repo/
  * https://repo.maven.apache.org/maven2/

It's probably a bug that it's not in the apache repo, but still it got to the `central.maven.org` repo. Any way to add repos to search in like we can do in maven?

**Comment by ahgittin** [comment]
@neykov can you confirm nothing wrong dropping to guava 16?

**Comment by ahgittin** [comment]
also note the guava version changes here conflicts with the swagger guava addition; trivial to merge, done in https://github.com/ahgittin/brooklyn-server/commit/100d6d48206f74ea43f0921d8844989318c05ad1

**Comment by neykov** [comment]
Haven't seen any problems with downgrading to guava 16. Tests are working fine as well.
Could look in emails around the time of change for potential rebind problems.

**Comment by neykov** [comment]
Looking at the changes now I see I've forgotten to move okhttp's version to pom.xml. Will do it in another commit not to mess your merge. Or if you are doing any changes do it in the branch directly to avoid messing your merge.

**Comment by ahgittin** [comment]
you can push it here, and ping me and i'll cherry-pick it (or open against my branch)

**Comment by neykov** [comment]
Moved versions here, commit bd91628.

**Pull request closed without merging**